test.html

```diff
@@ -164,6 +281,65 @@ <h2>Test drive</h2>
 		});
 }
 
+/**
+ * Loads a plugin including it css file
+ *
+ * @param  {string} plugin the plugin to load
+ * @type {Promise} the promise which resolves as soon as plugin is loaded
+ */
+function loadPlugin (plugin)
+{
+	if (!components.plugins[plugin]){
+		return;
+	}
+
+	var pluginPath = components.plugins.meta.path;
+
+	return new Promise(function (resolve) {
+		pluginPath = pluginPath.replace(/{id}/g, plugin);
+
+		$u.script(pluginPath + '.js', function() {
+			this.setAttribute('data-type', 'plugin');
+
+			var style = document.createElement('link');
+			style.setAttribute('rel', 'stylesheet');
+			style.setAttribute('href', pluginPath + '.css')
+			style.setAttribute('data-type', 'plugin');
+			document.body.appendChild(style);
+
+			resolve();
+		});
+	});
+}
+
+/**
+ * Reload all selected plugins and prism core
+ * You can hit me for such solution, but there are no other way :)
+ */
+function reloadPlugins ()
+{
+	var selectedPlugins = $$('input[type="checkbox"]:checked', document.forms[0]);
+	selectedPlugins = selectedPlugins.map(function(plugin) { return plugin.value });
+
+	$('#prism').remove();
+	$$('script[data-type="plugin"],link[data-type="plugin"]').forEach(function (script) {
+		script.remove();
+	});
+
+	// firstly loads prism core, after it - all selected plugins
+	return new Promise(function (resolve) {
+		$u.script('prism.js', function () {
+			this.setAttribute('id', 'prism');
+
```

Comment:
We should `resolve` the `Promise` here and `return Promise.all([...])` within the `then` handler. It'll be a little cleaner.

Reply:
Something like:
```javascript
return new Promise(function (resolve) {
  $u.script('prism.js', function () { 
    this.setAttribute('id', 'prism');
    resolve();
  });
}).then(function() {
  return Promise.all(selectedPlugins.map());
});
```
?

Reply:
Yeah, that's the idea.